tests/test_table.py

```diff
@@ -126,22 +126,37 @@ def test_query(TestModel, TestModel_entries, dynamo_local):
 
 def test_scan(TestModel, TestModel_entries, dynamo_local):
     """Scanning should return the expected values"""
-    results = TestModel.scan(count__gt=200)
+    results = list(TestModel.scan(count__gt=200))
     assert len(results) == 2
 
     # our table has a hash and range key, so our results are ordered based on the range key
     assert results[0].count == 333
     assert results[1].count == 222
 
-    results = TestModel.scan(child__sub="two")
+    results = list(TestModel.scan(child__sub="two"))
     assert len(results) == 1
     assert results[0].count == 222
 
-    results = TestModel.scan(child__sub__begins_with="t")
+    results = list(TestModel.scan(child__sub__begins_with="t"))
     assert len(results) == 2
     assert results[0].count == 333
     assert results[1].count == 222
 
+    # Test auto-paging
+    results = list(TestModel.scan(scan_kwargs={'Limit': 2}, count__gt=0))
+    assert len(results) == 3
```

Comment:
My thoughts: I'll accept `Limit`, and pass off to a helper function that calls table.scan and decrements the limit by `ScannedCount` for each call, thus simulating respecting the limit if the limit is greater than a 1mb page.